src/weighted-api/algo.spec.ts

```diff
@@ -0,0 +1,161 @@
+import { getRankingDistribution } from './algo';
+import { exportedForTesting } from './algo';
+
```

Comment:
lets add a test case where we ensure total sum of the final distribution is always 1.